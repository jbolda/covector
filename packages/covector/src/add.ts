import { type Logger } from "@covector/types";
import inquirer from "inquirer";
import { default as fsDefault } from "fs";
// this is compatible with node@12+
const fs = fsDefault.promises;
import { join } from "path";
import { configFile } from "@covector/files";

import type { ConfigFile } from "@covector/types";

export const add = function* ({
  logger,
  cwd = process.cwd(),
  changeFolder = ".changes",
  yes,
}: {
  logger: Logger;
  cwd?: string;
  changeFolder?: string;
  yes: boolean;
}): Generator<any, string, any> {
  const config: ConfigFile = yield configFile({ cwd });
  let packageBumps: { [k: string]: { bump: string; changeTag?: string } } = {};

  const answers: { packages: string[] } = yield inquirer.prompt([
    {
      type: "checkbox",
      message: "Select packages which need a version bump.",
      name: "packages",
      choices: Object.keys(config.packages).map((pkg) => ({ name: pkg })),
      validate(answer) {
        if (answer.length < 1) {
          return "You must choose at least one package.";
        }
        return true;
      },
    },
  ]);

  for (let pkg of answers.packages) {
    const { bump } = yield inquirer.prompt({
      type: "list",
      message: `bump ${pkg} with?`,
      name: "bump",
      choices: ["patch", "minor", "major"].concat(
        config.additionalBumpTypes ? config.additionalBumpTypes : []
      ),
    });
    let changeTag;
    if (config?.changeTags) {
      const tags = Object.keys(config.changeTags);
      const addTag = yield inquirer.prompt({
        type: "list",
        name: "changeTag",
        message: `bump ${pkg} with?`,
        choices: ["none"].concat(tags),
      });
      if (addTag.changeTag !== "none") changeTag = addTag.changeTag;
    }
    packageBumps[pkg] = { bump, changeTag };
  }

  const summary = yield inquirer.prompt({
    type: "input",
    name: "input",
    message: `Please summarize the changes that occurred.`,
    validate(answer) {
      if (answer.length < 1) {
        return "You must enter a summary.";
      }
      return true;
    },
  });

  const file = yield inquirer.prompt({
    type: "input",
    name: "name",
    message: `Please name the change file.`,
    validate(answer) {
      if (answer.length < 1) {
        return "You must enter a file name.";
      }
      return true;
    },
    filter: (input) => {
      if (input.endsWith(".md")) {
        return input;
      } else {
        return `${input}.md`;
      }
    },
  });

  const frontmatter = `---
${answers.packages
  .map(
    (pkg) =>
      `"${pkg}": ${packageBumps[pkg].bump}${packageBumps[pkg].changeTag ? `:${packageBumps[pkg].changeTag}` : ``}`
  )
  .join("\n")}
---\n\n`;

  const content = `${frontmatter}${summary.input}\n`;

  yield fs.writeFile(join(cwd, changeFolder, `${file.name}`), content);

  return "complete";
};
