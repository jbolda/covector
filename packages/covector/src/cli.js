const yargs = require("yargs");
const { init } = require("./init.js");

module.exports.cli = function* (argv, covector) {
  const options = parseOptions(argv);
  if (options.command === "init")
    return yield init({ ...options, changeFolder: options.directory });
  return yield covector(options);
};

function parseOptions(argv) {
  let rawOptions = yargs({})
    .scriptName("covector")
    .command("init", "initialize covector in your repo", function (yargs) {
      return yargs
        .option("yes", {
          alias: "y",
          describe: "skip all questions using the default response for each",
          default: false,
        })
        .boolean("yes")
        .option("directory", {
          alias: "d",
          describe: "specify a custom change folder",
          hidden: true,
        });
    })
    .command(["status", "*"], "run status command")
    .command("config", "output current config")
    .command("version", "run version command")
    .command("publish", "run publish command")
    .option("dry-run", {
      describe:
        "run a command that shows the expected command without executing",
    })
    .demandCommand(1)
    .help()
    .epilogue(
      "For more information on covector, see: https://www.github.com/jbolda/covector"
    )
    .parse(argv);

  return {
    command: rawOptions._[0],
    dryRun: rawOptions.dryRun,
    yes: rawOptions.yes,
    directory: rawOptions.directory,
  };
}
