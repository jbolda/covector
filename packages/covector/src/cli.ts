import yargs from "yargs";
import { covector } from "./run";
import { pino } from "pino";
import logStream from "./logger";

export function* cli(argv: readonly string[]): Generator<any, any, any> {
  const { command, directory, yes, dryRun, cwd } = parseOptions(argv);
  const stream = logStream();
  const logger = pino(stream);
  return yield covector({
    logger,
    command,
    changeFolder: directory,
    yes,
    dryRun,
    cwd,
  });
}

function parseOptions(argv: readonly string[]): {
  command: string;
  dryRun: boolean;
  yes?: boolean;
  directory?: string;
  cwd: string;
} {
  let rawOptions = yargs
    .scriptName("covector")
    .command("init", "initialize covector in your repo", function (yargs) {
      return yargs
        .option("yes", {
          alias: "y",
          describe: "skip all questions using the default response for each",
          default: false,
        })
        .boolean("yes")
        .option("directory", {
          alias: "d",
          describe: "specify a custom change folder",
          default: ".changes",
          hidden: true,
        });
    })
    .command(["status", "*"], "run status command")
    .command("config", "output current config")
    .command("version", "run version command")
    .command("publish", "run publish command")
    .options({
      "dry-run": {
        type: "boolean",
        default: false,
        describe:
          "run a command that shows the expected command without executing",
      },
    })
    .options({
      cwd: {
        type: "string",
        default: ".",
        describe: "context in which to run",
      },
    })
    .demandCommand(1)
    .help()
    .epilogue(
      "For more information on covector, see: https://www.github.com/jbolda/covector"
    )
    .parse(argv);

  // TODO type narrow, it thinks it could be a promise
  return {
    // @ts-expect-error
    command: String(rawOptions._[0]),
    // @ts-expect-error
    dryRun: rawOptions["dry-run"],
    // @ts-expect-error
    yes: rawOptions.yes,
    // @ts-expect-error
    directory: rawOptions.directory,
    // @ts-expect-error
    cwd: rawOptions.cwd,
  };
}
