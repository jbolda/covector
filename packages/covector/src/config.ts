import { type Logger } from "@covector/types";
import { configFile } from "@covector/files";

import type { Covector } from "@covector/types";

export function* config({
  logger,
  cwd = process.cwd(),
  modifyConfig = async (c) => c,
}: {
  logger: Logger;
  cwd?: string;
  modifyConfig?: (c: any) => Promise<any>;
}): Generator<any, Covector, any> {
  let config = yield modifyConfig(yield configFile({ cwd }));
  delete config.file;
  logger.info({ renderAsYAML: config });
  return { response: "config returned" };
}
