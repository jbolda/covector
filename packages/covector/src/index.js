const { covector } = require("./run");
module.exports.covector = covector;

const { cli } = require("./cli");
module.exports.cli = cli;
