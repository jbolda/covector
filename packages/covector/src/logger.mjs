import build from "pino-abstract-transport";
import yaml from "js-yaml";

const logLevel = {
  10: "trace",
  20: "debug",
  30: "info",
  40: "warn",
  50: "error",
  60: "fatal",
};

export default async function (opts) {
  return build(async function (source) {
    for await (let log of source) {
      console.dir(log);
      const level = `${logLevel[log.level].toUpperCase()}`;
      const command = log?.command ? `:${log.command}` : "";
      const covectorModule = log?.module ? `:${log.module}` : "";
      const msg = log?.msg ? `:: ${log.msg}` : "";
      const renderAsYAML = log?.renderAsYAML
        ? `\n    ${yaml.dump(log.renderAsYAML).replace(/\n/gm, "\n    ")}`
        : "";

      console.log(
        `${level}|${log.name}${command}${covectorModule} ${msg}${renderAsYAML}`
      );
    }
  });
}
