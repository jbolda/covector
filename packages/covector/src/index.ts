import { covector } from "./run";
import { cli } from "./cli";

export { covector, cli };
