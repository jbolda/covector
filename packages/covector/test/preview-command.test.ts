import { covector } from "../src";
import { it } from "@effection/jest";
import mockConsole from "jest-mock-console";
import fixtures from "fixturez";
import { TomlDocument } from "@covector/toml";
const f = fixtures(__dirname);

expect.addSnapshotSerializer({
  test: (value) => value instanceof TomlDocument,
  print: (_) => `TomlDocument {}`,
});

describe("integration test for preview command", () => {
  let restoreConsole: Function;
  beforeEach(() => {
    restoreConsole = mockConsole(["log", "dir", "info", "error"]);
  });
  afterEach(() => {
    restoreConsole();
  });

  it("runs version and publish for js and rust", function* () {
    const fullIntegration = f.copy("integration.js-and-rust-for-preview");
    const covectored = yield covector({
      command: "preview",
      cwd: fullIntegration,
      previewVersion: "branch-name.12345",
    });

    expect({
      consoleLog: (console.log as any).mock.calls,
      consoleInfo: (console.info as any).mock.calls,
      covectorReturn: covectored,
    }).toMatchSnapshot();
  });
});

describe("integration test for preview command with dist tags", () => {
  let restoreConsole: Function;
  beforeEach(() => {
    restoreConsole = mockConsole(["log", "dir", "info", "error"]);
  });
  afterEach(() => {
    restoreConsole();
  });

  it("runs version and publish for js and rust", function* () {
    const fullIntegration = f.copy("integration.js-and-rust-for-preview");
    const covectored = yield covector({
      command: "preview",
      cwd: fullIntegration,
      previewVersion: "branch-name.12345",
      branchTag: "branch_name",
    });

    expect({
      consoleLog: (console.log as any).mock.calls,
      consoleInfo: (console.info as any).mock.calls,
      covectorReturn: covectored,
    }).toMatchSnapshot();
  });
});
