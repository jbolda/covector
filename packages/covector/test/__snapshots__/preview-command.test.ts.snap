// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test for preview command runs version and publish for js and rust 1`] = `
Object {
  "consoleInfo": Array [],
  "consoleLog": Array [
    Array [
      "bumping package-b with branch-name.12345 identifier to publish a preview",
    ],
    Array [
      "bumping package-a with branch-name.12345 identifier to publish a preview",
    ],
    Array [
      "bumping package-c with branch-name.12345 identifier to publish a preview",
    ],
    Array [
      "package-a [prepublish]: echo prepublishing package-a would happen here",
    ],
    Array [
      "prepublishing package-a would happen here",
    ],
    Array [
      "package-b [prepublish]: echo prepublishing package-b would happen here",
    ],
    Array [
      "prepublishing package-b would happen here",
    ],
    Array [
      "package-c [prepublish]: echo prepublishing package-c would happen here",
    ],
    Array [
      "prepublishing package-c would happen here",
    ],
    Array [
      "package-a [publish]: echo publishing --tag  would happen here",
    ],
    Array [
      "publishing --tag would happen here",
    ],
    Array [
      "package-b [publish]: echo publishing --tag  would happen here",
    ],
    Array [
      "publishing --tag would happen here",
    ],
    Array [
      "package-c [publish]: echo publishing would happen here",
    ],
    Array [
      "publishing would happen here",
    ],
  ],
  "covectorReturn": Object {
    "commandsRan": Object {
      "package-a": Object {
        "command": true,
        "pkg": Object {
          "command": Array [
            "echo publishing --tag  would happen here",
          ],
          "dependencies": Array [
            "package-b",
          ],
          "errorOnVersionRange": null,
          "manager": "javascript",
          "path": "./package-a",
          "pkg": "package-a",
          "pkgFile": Object {
            "currentVersion": "0.0.1-branch-name.12345",
            "deps": Object {
              "package-b": Array [
                Object {
                  "type": "dependencies",
                  "version": "0.0.1-branch-name.12345",
                },
              ],
            },
            "file": Object {
              "content": "{
  \\"name\\": \\"package-a\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {
    \\"package-b\\": \\"0.0.1-branch-name.12345\\"
  }
}
",
              "extname": ".json",
              "filename": "package",
              "path": "package-a/package.json",
            },
            "name": "package-a",
            "pkg": Object {
              "dependencies": Object {
                "package-b": "0.0.1-branch-name.12345",
              },
              "name": "package-a",
              "version": "0.0.1-branch-name.12345",
            },
            "version": "0.0.1-branch-name.12345",
            "versionMajor": 0,
            "versionMinor": 0,
            "versionPatch": 1,
            "versionPrerelease": Array [
              "branch-name",
              12345,
            ],
          },
          "postcommand": Array [
            "echo postpublishing package-a would happen here",
          ],
          "precommand": Array [
            "echo prepublishing package-a would happen here",
          ],
          "releaseTag": "package-a-v0.0.1-branch-name.12345",
        },
        "postcommand": false,
        "precommand": true,
        "published": true,
      },
      "package-b": Object {
        "command": true,
        "pkg": Object {
          "command": Array [
            "echo publishing --tag  would happen here",
          ],
          "dependencies": undefined,
          "errorOnVersionRange": null,
          "manager": "javascript",
          "path": "./package-b",
          "pkg": "package-b",
          "pkgFile": Object {
            "currentVersion": "0.0.1-branch-name.12345",
            "deps": Object {},
            "file": Object {
              "content": "{
  \\"name\\": \\"package-b\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {}
}
",
              "extname": ".json",
              "filename": "package",
              "path": "package-b/package.json",
            },
            "name": "package-b",
            "pkg": Object {
              "dependencies": Object {},
              "name": "package-b",
              "version": "0.0.1-branch-name.12345",
            },
            "version": "0.0.1-branch-name.12345",
            "versionMajor": 0,
            "versionMinor": 0,
            "versionPatch": 1,
            "versionPrerelease": Array [
              "branch-name",
              12345,
            ],
          },
          "postcommand": Array [
            "echo postpublishing package-b would happen here",
          ],
          "precommand": Array [
            "echo prepublishing package-b would happen here",
          ],
          "releaseTag": "package-b-v0.0.1-branch-name.12345",
        },
        "postcommand": false,
        "precommand": true,
        "published": true,
      },
      "package-c": Object {
        "command": true,
        "pkg": Object {
          "command": Array [
            "echo publishing would happen here",
          ],
          "dependencies": Array [
            "package-b",
          ],
          "errorOnVersionRange": null,
          "manager": "rust",
          "path": "./package-c",
          "pkg": "package-c",
          "pkgFile": Object {
            "currentVersion": "0.0.1-branch-name.12345",
            "deps": Object {
              "package-b": Array [
                Object {
                  "type": "dependencies",
                  "version": "0.0.1-branch-name.12345",
                },
              ],
            },
            "file": Object {
              "content": "[package]
name = \\"package-c\\"
version = \\"0.0.1-branch-name.12345\\"

[dependencies]
package-b = \\"0.0.1-branch-name.12345\\"
",
              "extname": ".toml",
              "filename": "Cargo",
              "path": "package-c/Cargo.toml",
            },
            "name": "package-c",
            "pkg": TomlDocument {},
            "version": "0.0.1-branch-name.12345",
            "versionMajor": 0,
            "versionMinor": 0,
            "versionPatch": 1,
            "versionPrerelease": Array [
              "branch-name",
              12345,
            ],
          },
          "postcommand": Array [
            "echo postpublishing package-c would happen here",
          ],
          "precommand": Array [
            "echo prepublishing package-c would happen here",
          ],
          "releaseTag": "package-c-v0.0.1-branch-name.12345",
        },
        "postcommand": false,
        "precommand": true,
        "published": true,
      },
    },
    "pipeTemplate": Array [
      Object {
        "command": Array [
          "echo publishing --tag  would happen here",
        ],
        "dependencies": Array [
          "package-b",
        ],
        "errorOnVersionRange": null,
        "manager": "javascript",
        "path": "./package-a",
        "pkg": "package-a",
        "pkgFile": Object {
          "currentVersion": "0.0.1-branch-name.12345",
          "deps": Object {
            "package-b": Array [
              Object {
                "type": "dependencies",
                "version": "0.0.1-branch-name.12345",
              },
            ],
          },
          "file": Object {
            "content": "{
  \\"name\\": \\"package-a\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {
    \\"package-b\\": \\"0.0.1-branch-name.12345\\"
  }
}
",
            "extname": ".json",
            "filename": "package",
            "path": "package-a/package.json",
          },
          "name": "package-a",
          "pkg": Object {
            "dependencies": Object {
              "package-b": "0.0.1-branch-name.12345",
            },
            "name": "package-a",
            "version": "0.0.1-branch-name.12345",
          },
          "version": "0.0.1-branch-name.12345",
          "versionMajor": 0,
          "versionMinor": 0,
          "versionPatch": 1,
          "versionPrerelease": Array [
            "branch-name",
            12345,
          ],
        },
        "postcommand": Array [
          "echo postpublishing package-a would happen here",
        ],
        "precommand": Array [
          "echo prepublishing package-a would happen here",
        ],
        "releaseTag": "package-a-v0.0.1-branch-name.12345",
      },
      Object {
        "command": Array [
          "echo publishing --tag  would happen here",
        ],
        "dependencies": undefined,
        "errorOnVersionRange": null,
        "manager": "javascript",
        "path": "./package-b",
        "pkg": "package-b",
        "pkgFile": Object {
          "currentVersion": "0.0.1-branch-name.12345",
          "deps": Object {},
          "file": Object {
            "content": "{
  \\"name\\": \\"package-b\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {}
}
",
            "extname": ".json",
            "filename": "package",
            "path": "package-b/package.json",
          },
          "name": "package-b",
          "pkg": Object {
            "dependencies": Object {},
            "name": "package-b",
            "version": "0.0.1-branch-name.12345",
          },
          "version": "0.0.1-branch-name.12345",
          "versionMajor": 0,
          "versionMinor": 0,
          "versionPatch": 1,
          "versionPrerelease": Array [
            "branch-name",
            12345,
          ],
        },
        "postcommand": Array [
          "echo postpublishing package-b would happen here",
        ],
        "precommand": Array [
          "echo prepublishing package-b would happen here",
        ],
        "releaseTag": "package-b-v0.0.1-branch-name.12345",
      },
      Object {
        "command": Array [
          "echo publishing would happen here",
        ],
        "dependencies": Array [
          "package-b",
        ],
        "errorOnVersionRange": null,
        "manager": "rust",
        "path": "./package-c",
        "pkg": "package-c",
        "pkgFile": Object {
          "currentVersion": "0.0.1-branch-name.12345",
          "deps": Object {
            "package-b": Array [
              Object {
                "type": "dependencies",
                "version": "0.0.1-branch-name.12345",
              },
            ],
          },
          "file": Object {
            "content": "[package]
name = \\"package-c\\"
version = \\"0.0.1-branch-name.12345\\"

[dependencies]
package-b = \\"0.0.1-branch-name.12345\\"
",
            "extname": ".toml",
            "filename": "Cargo",
            "path": "package-c/Cargo.toml",
          },
          "name": "package-c",
          "pkg": TomlDocument {},
          "version": "0.0.1-branch-name.12345",
          "versionMajor": 0,
          "versionMinor": 0,
          "versionPatch": 1,
          "versionPrerelease": Array [
            "branch-name",
            12345,
          ],
        },
        "postcommand": Array [
          "echo postpublishing package-c would happen here",
        ],
        "precommand": Array [
          "echo prepublishing package-c would happen here",
        ],
        "releaseTag": "package-c-v0.0.1-branch-name.12345",
      },
    ],
  },
}
`;

exports[`integration test for preview command with dist tags runs version and publish for js and rust 1`] = `
Object {
  "consoleInfo": Array [],
  "consoleLog": Array [
    Array [
      "bumping package-b with branch-name.12345 identifier to publish a preview",
    ],
    Array [
      "bumping package-a with branch-name.12345 identifier to publish a preview",
    ],
    Array [
      "bumping package-c with branch-name.12345 identifier to publish a preview",
    ],
    Array [
      "package-a [prepublish]: echo prepublishing package-a would happen here",
    ],
    Array [
      "prepublishing package-a would happen here",
    ],
    Array [
      "package-b [prepublish]: echo prepublishing package-b would happen here",
    ],
    Array [
      "prepublishing package-b would happen here",
    ],
    Array [
      "package-c [prepublish]: echo prepublishing package-c would happen here",
    ],
    Array [
      "prepublishing package-c would happen here",
    ],
    Array [
      "package-a [publish]: echo publishing would happen here",
    ],
    Array [
      "publishing would happen here",
    ],
    Array [
      "package-b [publish]: echo publishing would happen here",
    ],
    Array [
      "publishing would happen here",
    ],
    Array [
      "package-c [publish]: echo publishing --tag branch_name would happen here",
    ],
    Array [
      "publishing --tag branch_name would happen here",
    ],
  ],
  "covectorReturn": Object {
    "commandsRan": Object {
      "package-a": Object {
        "command": true,
        "pkg": Object {
          "command": Array [
            "echo publishing would happen here",
          ],
          "dependencies": Array [
            "package-b",
          ],
          "errorOnVersionRange": null,
          "manager": "javascript",
          "path": "./package-a",
          "pkg": "package-a",
          "pkgFile": Object {
            "currentVersion": "0.0.1-branch-name.12345",
            "deps": Object {
              "package-b": Array [
                Object {
                  "type": "dependencies",
                  "version": "0.0.1-branch-name.12345",
                },
              ],
            },
            "file": Object {
              "content": "{
  \\"name\\": \\"package-a\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {
    \\"package-b\\": \\"0.0.1-branch-name.12345\\"
  }
}
",
              "extname": ".json",
              "filename": "package",
              "path": "package-a/package.json",
            },
            "name": "package-a",
            "pkg": Object {
              "dependencies": Object {
                "package-b": "0.0.1-branch-name.12345",
              },
              "name": "package-a",
              "version": "0.0.1-branch-name.12345",
            },
            "version": "0.0.1-branch-name.12345",
            "versionMajor": 0,
            "versionMinor": 0,
            "versionPatch": 1,
            "versionPrerelease": Array [
              "branch-name",
              12345,
            ],
          },
          "postcommand": Array [
            "echo postpublishing package-a would happen here",
          ],
          "precommand": Array [
            "echo prepublishing package-a would happen here",
          ],
          "releaseTag": "package-a-v0.0.1-branch-name.12345",
        },
        "postcommand": false,
        "precommand": true,
        "published": true,
      },
      "package-b": Object {
        "command": true,
        "pkg": Object {
          "command": Array [
            "echo publishing would happen here",
          ],
          "dependencies": undefined,
          "errorOnVersionRange": null,
          "manager": "javascript",
          "path": "./package-b",
          "pkg": "package-b",
          "pkgFile": Object {
            "currentVersion": "0.0.1-branch-name.12345",
            "deps": Object {},
            "file": Object {
              "content": "{
  \\"name\\": \\"package-b\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {}
}
",
              "extname": ".json",
              "filename": "package",
              "path": "package-b/package.json",
            },
            "name": "package-b",
            "pkg": Object {
              "dependencies": Object {},
              "name": "package-b",
              "version": "0.0.1-branch-name.12345",
            },
            "version": "0.0.1-branch-name.12345",
            "versionMajor": 0,
            "versionMinor": 0,
            "versionPatch": 1,
            "versionPrerelease": Array [
              "branch-name",
              12345,
            ],
          },
          "postcommand": Array [
            "echo postpublishing package-b would happen here",
          ],
          "precommand": Array [
            "echo prepublishing package-b would happen here",
          ],
          "releaseTag": "package-b-v0.0.1-branch-name.12345",
        },
        "postcommand": false,
        "precommand": true,
        "published": true,
      },
      "package-c": Object {
        "command": true,
        "pkg": Object {
          "command": Array [
            "echo publishing --tag branch_name would happen here",
          ],
          "dependencies": Array [
            "package-b",
          ],
          "errorOnVersionRange": null,
          "manager": "rust",
          "path": "./package-c",
          "pkg": "package-c",
          "pkgFile": Object {
            "currentVersion": "0.0.1-branch-name.12345",
            "deps": Object {
              "package-b": Array [
                Object {
                  "type": "dependencies",
                  "version": "0.0.1-branch-name.12345",
                },
              ],
            },
            "file": Object {
              "content": "[package]
name = \\"package-c\\"
version = \\"0.0.1-branch-name.12345\\"

[dependencies]
package-b = \\"0.0.1-branch-name.12345\\"
",
              "extname": ".toml",
              "filename": "Cargo",
              "path": "package-c/Cargo.toml",
            },
            "name": "package-c",
            "pkg": TomlDocument {},
            "version": "0.0.1-branch-name.12345",
            "versionMajor": 0,
            "versionMinor": 0,
            "versionPatch": 1,
            "versionPrerelease": Array [
              "branch-name",
              12345,
            ],
          },
          "postcommand": Array [
            "echo postpublishing package-c would happen here",
          ],
          "precommand": Array [
            "echo prepublishing package-c would happen here",
          ],
          "releaseTag": "package-c-v0.0.1-branch-name.12345",
        },
        "postcommand": false,
        "precommand": true,
        "published": true,
      },
    },
    "pipeTemplate": Array [
      Object {
        "command": Array [
          "echo publishing would happen here",
        ],
        "dependencies": Array [
          "package-b",
        ],
        "errorOnVersionRange": null,
        "manager": "javascript",
        "path": "./package-a",
        "pkg": "package-a",
        "pkgFile": Object {
          "currentVersion": "0.0.1-branch-name.12345",
          "deps": Object {
            "package-b": Array [
              Object {
                "type": "dependencies",
                "version": "0.0.1-branch-name.12345",
              },
            ],
          },
          "file": Object {
            "content": "{
  \\"name\\": \\"package-a\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {
    \\"package-b\\": \\"0.0.1-branch-name.12345\\"
  }
}
",
            "extname": ".json",
            "filename": "package",
            "path": "package-a/package.json",
          },
          "name": "package-a",
          "pkg": Object {
            "dependencies": Object {
              "package-b": "0.0.1-branch-name.12345",
            },
            "name": "package-a",
            "version": "0.0.1-branch-name.12345",
          },
          "version": "0.0.1-branch-name.12345",
          "versionMajor": 0,
          "versionMinor": 0,
          "versionPatch": 1,
          "versionPrerelease": Array [
            "branch-name",
            12345,
          ],
        },
        "postcommand": Array [
          "echo postpublishing package-a would happen here",
        ],
        "precommand": Array [
          "echo prepublishing package-a would happen here",
        ],
        "releaseTag": "package-a-v0.0.1-branch-name.12345",
      },
      Object {
        "command": Array [
          "echo publishing would happen here",
        ],
        "dependencies": undefined,
        "errorOnVersionRange": null,
        "manager": "javascript",
        "path": "./package-b",
        "pkg": "package-b",
        "pkgFile": Object {
          "currentVersion": "0.0.1-branch-name.12345",
          "deps": Object {},
          "file": Object {
            "content": "{
  \\"name\\": \\"package-b\\",
  \\"version\\": \\"0.0.1-branch-name.12345\\",
  \\"dependencies\\": {}
}
",
            "extname": ".json",
            "filename": "package",
            "path": "package-b/package.json",
          },
          "name": "package-b",
          "pkg": Object {
            "dependencies": Object {},
            "name": "package-b",
            "version": "0.0.1-branch-name.12345",
          },
          "version": "0.0.1-branch-name.12345",
          "versionMajor": 0,
          "versionMinor": 0,
          "versionPatch": 1,
          "versionPrerelease": Array [
            "branch-name",
            12345,
          ],
        },
        "postcommand": Array [
          "echo postpublishing package-b would happen here",
        ],
        "precommand": Array [
          "echo prepublishing package-b would happen here",
        ],
        "releaseTag": "package-b-v0.0.1-branch-name.12345",
      },
      Object {
        "command": Array [
          "echo publishing --tag branch_name would happen here",
        ],
        "dependencies": Array [
          "package-b",
        ],
        "errorOnVersionRange": null,
        "manager": "rust",
        "path": "./package-c",
        "pkg": "package-c",
        "pkgFile": Object {
          "currentVersion": "0.0.1-branch-name.12345",
          "deps": Object {
            "package-b": Array [
              Object {
                "type": "dependencies",
                "version": "0.0.1-branch-name.12345",
              },
            ],
          },
          "file": Object {
            "content": "[package]
name = \\"package-c\\"
version = \\"0.0.1-branch-name.12345\\"

[dependencies]
package-b = \\"0.0.1-branch-name.12345\\"
",
            "extname": ".toml",
            "filename": "Cargo",
            "path": "package-c/Cargo.toml",
          },
          "name": "package-c",
          "pkg": TomlDocument {},
          "version": "0.0.1-branch-name.12345",
          "versionMajor": 0,
          "versionMinor": 0,
          "versionPatch": 1,
          "versionPrerelease": Array [
            "branch-name",
            12345,
          ],
        },
        "postcommand": Array [
          "echo postpublishing package-c would happen here",
        ],
        "precommand": Array [
          "echo prepublishing package-c would happen here",
        ],
        "releaseTag": "package-c-v0.0.1-branch-name.12345",
      },
    ],
  },
}
`;
